### PR TITLE
Postgres Parameters Page

### PR DESCRIPTION
**Pull request by nasc17**
Getting postgres parameters page. Users can perform actions such as connection to PostgresSQL server, viewing engine settings with their current values, searching through rows, setting server parameters, and resetting parameters to default values,

initial

UI is currently freezing when trying to populate large amount of server parameters in the declarative table. Bug issued here: https://github.com/microsoft/azuredatastudio/issues/13905

### CODE REVIEW

File: extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+		content.addItem(infoAndLink, { CSSStyles: { 'margin-bottom': '20px' } });
+
+		content.addItem(this.searchBox!, { CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px', 'margin-bottom': '20px' } });
+
+		this.parametersTable = this.modelView.modelBuilder.declarativeTable().withProps({
+			width: '100%',
+			columns: [
+				{
+					displayName: 'Parameter Name',
+					valueType: azdata.DeclarativeDataType.component,
```
Comment (bergeron) [inline extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts:89]: Can the name and description columns be strings rather than components? i.e. `azdata.DeclarativeDataType.string`

File: extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts [hunk 2]
```diff
@@ -290,211 +290,261 @@ export class PostgresParametersPage extends DashboardPage {
...
-		"shared_preload_libraries", "synchronous_commit", "ssl", "unix_socket_permissions", "wal_level" */
-
-		// For ev in this._postgresModel._engineSettings
-		// create row
-		// return rows
-		this.parameterComponents('engineSetting', '');
-		return [];
+	private createParameters(): void {
+		this._parameters = [];
+		this._postgresModel._engineSettings.forEach(engineSetting => {
```
Comment (Charles-Gagnon) [inline extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts:354]: `this._parameters = this._postgresModel._engineSettings.map(engineSetting => this.createParameterComponents(engineSetting));`

File: extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts [hunk 3]
```diff
@@ -503,19 +553,39 @@ export class PostgresParametersPage extends DashboardPage {
...
+			this.resetAllButton!.enabled = true;
 			this.parameterContainer!.addItem(this.parametersTable!);
+			this.refreshParametersTable();
 		}
 	}
 
-	private handleEngineSettingsUpdated(): void {
-		//TODO
+	private async callGetEngineSettings(): Promise<void> {
+		await this._postgresModel.getEngineSettings().catch(err => {
```
Comment (Charles-Gagnon) [inline extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts:564]: Use try/catch instead of catch callbacks

File: extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts [hunk 4]
```diff
@@ -503,19 +553,39 @@ export class PostgresParametersPage extends DashboardPage {
...
+			// If an error occurs show a message so the user knows something failed but still
```
Comment (Charles-Gagnon) [inline extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts:565]: This comment doesn't seem to match what's going on here

**Review by Charles-Gagnon — APPROVED** [review]

File: resources/localization/LCL/fr/cms.xlf.lcl [hunk 5]
```diff
@@ -78,15 +78,6 @@
         </Str>
         <Disp Icon="Str" />
       </Item>
-      <Item ItemId=";extensions/cms/out/cmsResource/cmsResourceCommands.cms.errors.azureNotAllowed" ItemType="0" PsrId="308" Leaf="true">
```
Comment (bergeron) [inline resources/localization/LCL/fr/cms.xlf.lcl:81]: Undo this change?

File: extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts [hunk 6]
```diff
@@ -166,20 +155,19 @@ export class PostgresParametersPage extends DashboardPage {
 							cancellable: false
 						},
 						async (_progress, _token): Promise<void> => {
-							//Edit multiple
-							// azdata arc postgres server edit -n <server group name> -e '<parameter name>=<parameter value>, <parameter name>=<parameter value>,...'
 							try {
-								this.engineSettingUpdates!.forEach((value: string) => {
-									this.engineSettings += value + ', ';
+								this.parameterUpdates!.forEach((value, key) => {
+									engineSettings.push(`${key}=${value}`);
```
Comment (bergeron) [inline extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts:160]: It would be good to verify the values are escaped correctly.  Can we set settings whose values contain spaces, equals, commas?

Comment (bergeron) [inline extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts:160]: I thought --engine-settings accepted a comma separated list. Are the settings comma separated somewhere?

File: extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts [hunk 7]
```diff
@@ -290,232 +290,293 @@ export class PostgresParametersPage extends DashboardPage {
...
-		// return rows
-		this.parameterComponents('engineSetting', '');
-		return [];
+		this._parameters.forEach(param => {
+			if (param.parameterName?.search(search) !== -1 || param.description?.search(search) !== -1) {
+				filterData.push(param);
+			}
+		});
+
+		this.parametersTable.data = filterData.map(f => [f.parameterName, f.valueContainer, f.description, f.resetButton]);
```
Comment (bergeron) [inline extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts:379]: Could do something like this.  .filter() is a little easier to read than creating a new array and pushing to it.

```
this.parametersTable.data = this._parameters
    .filter(p => p.parameterName?.search(search) !== -1 || p.description?.search(search) !== -1)
    .map(p => [p.parameterName, p.valueContainer, p.description, p.resetButton])
```

File: extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts (hunk 6 above)
Comment (nasc17) [reply extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts:160]: The array uses the toString method when passing into the command to show as comma separated.

Comment (nasc17) [reply extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts]: Good call, will use quotation marks around the values for this.

**Review by bergeron — APPROVED** [review]
